### ISSUE TITLE
keyboard_enter_text doesn't work with numeric keyboards

### ISSUE THREAD

**Issue by louismza** [calabash/calabash-ios#1176]
With a numeric keyboard showing in the iOS simulator, when you run keyboard_enter_text it gives the following error:

```
keyboard_enter_text 5
HTTPClient::ReceiveTimeoutError: execution expired
    from /Users/louis/.rvm/gems/ruby-2.2.1/gems/httpclient-2.8.2.4/lib/httpclient/session.rb:799:in `gets'
    from /Users/louis/.rvm/gems/ruby-2.2.1/gems/httpclient-2.8.2.4/lib/httpclient/session.rb:799:in `block in parse_header'
    from /Users/louis/.rvm/rubies/ruby-2.2.1/lib/ruby/2.2.0/timeout.rb:89:in `block in timeout'
    from /Users/louis/.rvm/rubies/ruby-2.2.1/lib/ruby/2.2.0/timeout.rb:99:in `call'
    from /Users/louis/.rvm/rubies/ruby-2.2.1/lib/ruby/2.2.0/timeout.rb:99:in `timeout'
    from /Users/louis/.rvm/gems/ruby-2.2.1/gems/httpclient-2.8.2.4/lib/httpclient/session.rb:795:in `parse_header'
    from /Users/louis/.rvm/gems/ruby-2.2.1/gems/httpclient-2.8.2.4/lib/httpclient/session.rb:778:in `read_header'
    from /Users/louis/.rvm/gems/ruby-2.2.1/gems/httpclient-2.8.2.4/lib/httpclient/session.rb:554:in `get_header'
    from /Users/louis/.rvm/gems/ruby-2.2.1/gems/httpclient-2.8.2.4/lib/httpclient.rb:1297:in `do_get_header'
    from /Users/louis/.rvm/gems/ruby-2.2.1/gems/httpclient-2.8.2.4/lib/httpclient.rb:1243:in `do_get_block'
    from /Users/louis/.rvm/gems/ruby-2.2.1/gems/httpclient-2.8.2.4/lib/httpclient.rb:1017:in `block in do_request'
    from /Users/louis/.rvm/gems/ruby-2.2.1/gems/httpclient-2.8.2.4/lib/httpclient.rb:1131:in `protect_keep_alive_disconnected'
    from /Users/louis/.rvm/gems/ruby-2.2.1/gems/httpclient-2.8.2.4/lib/httpclient.rb:1012:in `do_request'
    from /Users/louis/.rvm/gems/ruby-2.2.1/gems/httpclient-2.8.2.4/lib/httpclient.rb:854:in `request'
    from /Users/louis/.rvm/gems/ruby-2.2.1/gems/httpclient-2.8.2.4/lib/httpclient.rb:763:in `post'
    from /Users/louis/.rvm/gems/ruby-2.2.1/gems/run_loop-2.2.0/lib/run_loop/http/retriable_client.rb:184:in `send_request'
    from /Users/louis/.rvm/gems/ruby-2.2.1/gems/run_loop-2.2.0/lib/run_loop/http/retriable_client.rb:163:in `block in request'
    from /Users/louis/.rvm/gems/ruby-2.2.1/gems/run_loop-2.2.0/lib/run_loop/http/retriable_client.rb:150:in `times'
    from /Users/louis/.rvm/gems/ruby-2.2.1/gems/run_loop-2.2.0/lib/run_loop/http/retriable_client.rb:150:in `request'
    from /Users/louis/.rvm/gems/ruby-2.2.1/gems/run_loop-2.2.0/lib/run_loop/http/retriable_client.rb:93:in `post'
    from /Users/louis/.rvm/gems/ruby-2.2.1/gems/run_loop-2.2.0/lib/run_loop/device_agent/client.rb:1367:in `_dismiss_springboard_alerts'
    from /Users/louis/.rvm/gems/ruby-2.2.1/gems/calabash-cucumber-0.20.0/lib/calabash-cucumber/map.rb:61:in `map'
    from /Users/louis/.rvm/gems/ruby-2.2.1/gems/calabash-cucumber-0.20.0/lib/calabash-cucumber/keyboard_helpers.rb:200:in `_query_wrapper'
    from /Users/louis/.rvm/gems/ruby-2.2.1/gems/calabash-cucumber-0.20.0/lib/calabash-cucumber/keyboard_helpers.rb:205:in `_query_for_keyboard'
    from /Users/louis/.rvm/gems/ruby-2.2.1/gems/calabash-cucumber-0.20.0/lib/calabash-cucumber/keyboard_helpers.rb:28:in `docked_keyboard_visible?'
    from /Users/louis/.rvm/gems/ruby-2.2.1/gems/calabash-cucumber-0.20.0/lib/calabash-cucumber/keyboard_helpers.rb:80:in `keyboard_visible?'
    from /Users/louis/.rvm/gems/ruby-2.2.1/gems/calabash-cucumber-0.20.0/lib/calabash-cucumber/keyboard_helpers.rb:88:in `expect_keyboard_visible!'
    from /Users/louis/.rvm/gems/ruby-2.2.1/gems/calabash-cucumber-0.20.0/lib/calabash-cucumber/core.rb:684:in `keyboard_enter_text'
    from (irb):17
    from /Users/louis/.rvm/rubies/ruby-2.2.1/bin/irb:11
```

**Comment by jmoody** [comment]
Please use GitHub Markdown when posting code or stack traces.  Specifically, use "```".

**Comment by louismza** [comment]
@jmoody Sorry, will next time.

**Comment by kinejara** [comment]
This is also happening to me, I'm using the iOS 10 iPhone 7 simulator.

**Comment by jmoody** [comment]
We are working on adding tests for interacting with Numeric keyboards.

In the meantime, can you try `set_text`?

**Comment by mattspeers** [comment]
Same here, I just change to .to_s and then works fine

**Comment by jmoody** [comment]
@jescriba This can be fixed by us in run-loop:

https://github.com/calabash/run_loop/blob/develop/lib/run_loop/device_agent/client.rb#L365

by calling `.to_s` on the string argument.

**Comment by jmoody** [comment]
Fix will be available in run-loop 2.2.3
